infra/modules/azure_naming_convention/locals.tf

```diff
@@ -7,10 +7,81 @@ locals {
     "spaincentral"       = "spc"
   }
 
-  location_short = lookup(local.location_map, var.environment.location, "neu")
-  project        = "${var.environment.prefix}-${var.environment.env_short}-${local.location_short}"
-  domain         = var.environment.domain == null ? "-" : "-${var.environment.domain}-"
+  configurations = {
+    for env in var.environments : join("", [
+      "${env.prefix}-${env.env_short}-${lookup(local.location_map, env.location, "neu")}",
+      env.domain == null ? "-" : "-${env.domain}-",
+      env.app_name
+      ]) => {
+      prefix          = env.prefix
+      env_short       = env.env_short
+      location        = env.location
+      domain          = env.domain
+      app_name        = env.app_name
+      instance_number = env.instance_number
+      project         = "${env.prefix}-${env.env_short}-${lookup(local.location_map, env.location, "neu")}"
+      app_suffix      = [for i in range(env.instance_number) : format("%02d", i + 1)]
+    }
+  }
+
+  # Map resource types to their abbreviations
+  # Ref.: https://learn.microsoft.com/en-us/azure/cloud-adoption-framework/ready/azure-best-practices/resource-abbreviations
+  resource_abbreviations = {
+    # Compute
+    "virtual_machine" = "vm"
+
+    # Storage
+    "storage_account"        = "st"
+    "blob_storage"           = "blob"
+    "queue_storage"          = "queue"
+    "table_storage"          = "table"
+    "file_storage"           = "file"
+    function_storage_account = "stfn"
+
+    # Networking
+    "dns_zone"                    = "dns"
```

Comment:
Don't think this is used as the DNS zone has only the domain in the name?

Reply:
yes the dns name are different, I don't think it's necessary here, I'll remove it 👍